src/orion/core/worker/primary_algo.py

```diff
@@ -1,356 +1,109 @@
-        as a response to the running environment. Default is None response.
-
-        """
-        self._verify_trial(trial)
-        return self.algorithm.judge(
-            self.transformed_space.transform(trial), measurements
-        )
-
-    def should_suspend(self, trial: Trial) -> bool:
-        """Allow algorithm to decide whether a particular running trial is still
-        worth to complete its evaluation, based on information provided by the
-        `judge` method.
-
-        """
-        self._verify_trial(trial)
-        return self.algorithm.should_suspend(trial)
-
-    @property
-    def configuration(self) -> dict:
-        """Return tunable elements of this algorithm in a dictionary form
-        appropriate for saving.
-        """
-        # TODO: Return a dict with the wrapped algo's configuration instead?
-        # return {
-        #     type(self).__qualname__: {
-        #         "space": self.space.configuration,
-        #         "algorithm": {self.algorithm.configuration},
-        #     }
-        # }
-        return self.algorithm.configuration
-
-    @property
-    def space(self) -> Space:
-        """Domain of problem associated with this algorithm's instance.
-
-        .. note:: Redefining property here without setter, denies base class' setter.
-        """
-        return self._space
-
-    @property
-    def fidelity_index(self) -> str | None:
-        """Compute the index of the space where fidelity is.
-
-        Returns None if there is no fidelity dimension.
-        """
-        return self.algorithm.fidelity_index
-
-    def _verify_trial(self, trial: Trial, space: Optional[Space] = None) -> None:
-        if space is None:
-            space = self.space
-
-        if trial not in space:
-            raise ValueError(
-                f"Trial {trial.id} not contained in space:"
-                f"\nParams: {trial.params}\nSpace: {space}"
-            )
-
-
-def _copy_status_and_results(original_trial: Trial, transformed_trial: Trial) -> Trial:
-    """Copies the results, status, and other data from `transformed_trial` to `original_trial`."""
-    new_transformed_trial = copy.deepcopy(original_trial)
-    # pylint: disable=protected-access
-    new_transformed_trial._params = copy.deepcopy(transformed_trial._params)
-    return new_transformed_trial
+    spaces = [space]
+    # Create the spaces for each wrapper, from the top down.
+    if knowledge_base:
+        space = MultiTaskWrapper.transform_space(space, knowledge_base=knowledge_base)
+        spaces.append(space)
+
+    space = InsistSuggest.transform_space(space)
+    spaces.append(space)
+
+    space = SpaceTransform.transform_space(space, algo_type=algo_type)
+    spaces.append(space)
+
+    # Create the algo, using the innermost (most transformed) space.
+    # Then, create each wrapper, from the bottom up.
+    algorithm = algo_type(space=spaces.pop(), *algo_args, **algo_kwargs)
```

Comment:
Do I understand correctly that the algorithm's suggested trial, will first go through SpaceTransform, then InsistSuggest and finally MultiTaskWrapper?

Reply:
Indeed, that's the idea!